src/rgen/Microsoft.Macios.Generator/Emitters/BindingSyntaxFactory.Trampoline.cs

```diff
@@ -130,7 +130,7 @@ internal static TypeSyntax GetLowLevelType (in TypeInfo typeInfo)
 	/// </summary>
 	/// <param name="parameter">The parameter we want to generate for the lower invoke method.</param>
 	/// <returns>The parameter syntax needed for the parameter.</returns>
-	internal static ParameterSyntax GetTrampolineInvokeParameter (in DelegateParameter parameter)
+	internal static (SyntaxToken ParameterName, TypeSyntax ParameterType) GetTrampolineInvokeParameter (in DelegateParameter parameter)
```

Comment:
I was pondering about leaving a comment about not using a tuple as a return value, but that's internal, using out params isn't better, tuples elements are named... so yeah, no, that's good as is

Reply:
It is one of those things I also have mixed feelings. Had you asked me to changed it, I would have done it. I can always use a record, but seems  to be an overkill.